src/main/java/com/library/repository/postgres/BookRepository.java

```diff
@@ -13,4 +13,8 @@ public interface BookRepository extends JpaRepository<Book, Long> {
     @Modifying
     @Query("UPDATE Book b SET b.deleted = true WHERE b.id = :id")
     void delete(@Param(value = "id") Long bookId);
-}
+
+    @Modifying
+    @Query("DELETE FROM Book b WHERE b.id = :id")
+    void hardDelete(@Param(value = "id") Long bookId);
```

Comment:
naming:

delete - is soft delete.
New developer should understand just by name what method do.

hard delete - is default deleteById.
Dont create extra methods.

Reply:
Ok